lib/aux_vis.cpp

```diff
@@ -1783,3 +1783,38 @@ void SetFont(const std::string& fn)
         << "', height = " << font_size << endl;
 #endif
 }
+
+function<string(double)> NumberFormatter(int precision, char format, bool showsign) {
+   return [precision, format, showsign](double x) -> string {
+      ostringstream oss;
+      switch (format) {
+         case 'f':
+               oss << fixed;
+               break;
+         case 's':
+               oss << scientific;
+               break;
+         case 'd':
+               oss << defaultfloat;
+               break;
+         default:
+            MFEM_WARNING("Unknown formatting type. Using default. "
+            << "Valid options include: ['f', 's', 'd']" << endl);
+            oss << defaultfloat;
+            break;
+      };
+      if (showsign){
+         oss << showpos;
+      }
+      oss << setprecision(precision) << x;
+      return oss.str();
+   };
+}
+
+function<string(double)> NumberFormatter(string formatting) {
+   return [formatting](double x) -> string {
+      char buf[64];
+      snprintf(buf, sizeof(buf), formatting.c_str(), x);
```

Comment:
Well, this is a really straightforward solution 😁 It is a pretty big security hole 💣 , but we are not developing some pentagon software here, so maybe ok. But someone may compromise maybe the [online version](https://glvis.org/live/), so I would add at least a basic sanitation of the string to check there are expected format characters 👀 .

Reply:
I don't expect `snprintf` to be a security risk, even with intentionally ill-formed formatting string -- you may get garbage in the output buffer but that should not be a problem.

Reply:
You may read memory through it, because it may instruct `snprintf` interpret the second argument as a pointer (to string). But I am suggesting to simply check if the formatting string contains characters we expect like `%+-.feg` and digits.

Reply:
By the way, if security is a concern (e.g. on machines with multiple users), you can build GLVis using MFEM built with GnuTLS. With such a build, socket connections will be authenticated on both sides, the client and the server.

Reply:
I'm glad you pointed this out! I added a regex check which is fairly restrictive. https://github.com/GLVis/glvis/pull/288/commits/f91f3e47217ee5700d3ce833767d9e22b27ef9a0

Reply:
Thanks, that is more than I was asking for 💯 😉 Maybe just why is there an apostrophe? 🤔

Reply:
🤫